addons/common/functions/fnc_registerAmmoFiredEvent.sqf

```diff
@@ -0,0 +1,54 @@
+/*
+ * Author: esteldunedain
+ * Register an ammo fired event that should be launched when ammo following a
+ * given criteria is fired. The criteria is analyzed on mission start and is
+ * cached.
+ *
+ * Arguments:
+ * 0: Name of the event <STRING>
+ * 1: Condition code <CODE>
+ * 2: Call for the local player on foot? <BOOL>
+ * 3: Call for non local players on foot? <BOOL>
+ * 4: Call for non players on foot? <BOOL>
+ * 5: Call for the local player vehicle? <BOOL>
+ * 6: Call for non local players vehicles? <BOOL>
+ * 7: Call for non players vehicles? <BOOL>
```

Comment:
> "Call for non players on foot?". Does that mean: "fire for all AIs, local or not"?

Yes.

> Does "on foot" mean that it does not trigger for FFV slots? 

I'm actually not sure about that one.  The check is if firedEH unit is of type CAManBase, which I think should be FFV compatible. 

-------

You should keep in mind that this is just trying to mimic the Unified Fire EH logic of dividing events in six categories:
- ace_firedPlayer
- ace_firedPlayerNonLocal
- ace_firedNonPlayer
- ace_firedPlayerVehicle
- ace_firedPlayerVehicleNonLocal
- ace_firedNonPlayerVehicle